DefaultCredentialsProvider is broken for java8 app engine

This was reported by a bigtable customer:

> I've traced through where it fails. In DefaultCredentialsProvider.getDefaultCredentialsUnsynchronized() it will only return the AppEngine credentials if the following evaluates to true:
> 
>     this.isOnGAEStandard7()
> 
> I am on GAE Standard but am using Java 8 not 7, so this check is false and it does not return any valid credentials.

I didn't work :/

I configured the environment variable through `appengine-web.xml` using the following  tag:

```xml
    <env-variables>
        <env-var name ="GCE_METADATA_HOST" value="metadata.google.internal"/>
    </env-variables>
```
The host doesn't need the protocol because the code already attributes it automatically:

```java
  public static String getMetadataServerUrl(DefaultCredentialsProvider provider) {
    String metadataServerAddress = provider.getEnv(DefaultCredentialsProvider.GCE_METADATA_HOST_ENV_VAR);
    if (metadataServerAddress != null) {
      return "http://" + metadataServerAddress;
    }
    return DEFAULT_METADATA_SERVER_URL;
  }
```
The default metadadata server url is `http://169.254.169.254`

I'm still getting the error

```
Caused by: java.io.IOException: The Application Default Credentials are not available. They are available if running in Google Compute Engine. Otherwise, the environment variable GOOGLE_APPLICATION_CREDENTIALS must be defined pointing to a file defining the credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.
	at com.google.auth.oauth2.DefaultCredentialsProvider.getDefaultCredentials(DefaultCredentialsProvider.java:119)
```

--- a/oauth2_http/java/com/google/auth/oauth2/ComputeEngineCredentials.java
+++ b/oauth2_http/java/com/google/auth/oauth2/ComputeEngineCredentials.java
@@ -71,7 +71,7 @@ public class ComputeEngineCredentials extends GoogleCredentials implements Servi
   private static final Logger LOGGER = Logger.getLogger(ComputeEngineCredentials.class.getName());
 
   // Note: the explicit IP address is used to avoid name server resolution issues.
-  static final String DEFAULT_METADATA_SERVER_URL = "http://169.254.169.254";
+  static final String DEFAULT_METADATA_SERVER_URL = "http://metadata.google.internal";
 
   static final String SIGN_BLOB_URL_FORMAT = "https://iam.googleapis.com/v1/projects/-/serviceAccounts/%s:signBlob?alt=json";
 

